olp-cpp-sdk-core/include/olp/core/cache/CacheSettings.h

```diff
@@ -150,6 +150,17 @@ struct CORE_API CacheSettings {
    * the network state.
    */
   boost::optional<std::string> disk_path_protected = boost::none;
+
+  /**
+   * @brief Sets the maximum portion of data that can be evicted from the mutable
+   * cache using single eviction call during decreasing maximum cache size.
+   * The larger this number, the more RAM will be needed for eviction. If bigger
+   * bunch of data must be evicted, eviction will be
+   * splitted to chunks with size limited to provided number.
+   *
+   * The default value is 128 MB.
+   */
+  uint64_t max_eviction_portion = 1024ull * 1024ull * 128ull;
```

Comment:
Why is this necessary? This is not a logical setting as you can only know how much to set this once you know how much you need to decrease the size of the cache and in most cases, you probably do not know this as it is given by the user. But you can not also change this value dynamically. Why don't you set it internally to 1MB or so and just evict and see how much it will take?